server/server_test.go

```diff
@@ -123,13 +123,15 @@ func (cli *testServerClient) runTestsOnNewDB(t *testing.T, overrider configOverr
 	dsn := cli.getDSN(overrider, func(config *mysql.Config) {
 		config.DBName = ""
 	})
+	cli.waitUntilCustomizationServerCanConnect(dsn)
```

Comment:
Is it possible that the client can connect to one DB but cannot connect to another? I don't think so.

Reply:
remove custom DSN and ```waitUntilCustomizationServerCanConnect```.

Reply:
Sorry, It still was needed. ```TestSocket``` use it to connect tidb by socket file